common/network.c

```diff
@@ -434,6 +434,23 @@ network_interface_is_wifi(const char *if_name)
 	return ret;
 }
 
+list_t *
+network_get_interfaces_new()
+{
+	struct if_nameindex *if_ni, *i;
+	if_ni = if_nameindex();
+
+	list_t *if_name_list = NULL;
+
+	IF_NULL_RETVAL(if_ni, NULL);
```

Comment:
i'd suggest moving this right after if_nameindex() to log errno on error, e.g. using IF_NULL_RETVAL_ERROR_ERRNO